src/main/scala/com/fulcrumgenomics/bam/SamRecordClipper.scala

```diff
@@ -248,6 +248,30 @@ class SamRecordClipper(val mode: ClippingMode, val autoClipAttributes: Boolean)
     else clipEndOfRead(rec, numberOfBasesToClip)
   }
 
+  /** Converts all soft-clipping to hard-clipping.  Does not convert unless the clipping mode is [[ClippingMode.Hard]].
+    * Calculates any clipping required and delegates to [[clipStartOfRead()]] and [[clipEndOfRead()]].
+    * @param rec the record to be clipped
+    * @return the number of bases converted to hard-clipping from soft-clipping at the start and end of the read
+    *         respectively.
+    */
+  def toHardClip(rec: SamRecord): (Int, Int) = {
```

Comment:
I really dislike how a) this method hard-codes the clipping type given that the rest of this class clips based on `ClippingMode`, and then b) refuses to do anything unless mode == Hard.

Reply:
I would suggest calling this method something like `upgradeAllClipping`, and letting it do it's thing in either Hard or SoftWithMask mode.

Reply:
Also, you return `(Int, Int)` here, but don't use that in `ClipBam`.  Is that intentional?